cmd/utils/flags.go

```
@@ -503,11 +503,15 @@ var (
 	}
 
 	// Anti-MEV settings.
-	AMEVKeystoreFlag = &cli.StringFlag{
-		Name:     "amev.keystore",
+	AntiMEVKeyStoreFlag = &cli.PathFlag{
+		Name:     "antimev.keystore",
 		Usage:    "Path to anti-MEV keystore",
-		Value:    "",
-		Category: flags.AMEVCategory,
+		Category: flags.AntiMEVCategory,
+	}
+	AntiMEVPasswordFlag = &cli.PathFlag{
+		Name:     "antimev.password",
+		Usage:    "Password file to unlock anti-MEV keystore",
+		Category: flags.AntiMEVCategory,
```

I didn't see it in the PR, but we need to add a requirement: if `--mine` flag is specified, i.e. the node is CN and dBFT engine is used, then it should be explicitly required from the user to specify AMEV keystore with password.

---

We have, ref https://github.com/bane-labs/go-ethereum/pull/326#discussion_r1791433169.

> if --mine flag is specified, i.e. the node is CN and dBFT engine is used, then it should be explicitly required from the user to specify AMEV keystore with password.

I'm not sure if we need another check, because `unlockAntiMEVKeyStore` will `utils.Fatalf` if the file is not found or the password is not correct, when `btf != nil`.

---

> because unlockAntiMEVKeyStore will utils.Fatalf if the file is not found

From what I see, there's a case when it will silently skip keystore initialisation if both file/password are not provided, which should be a no-op in case of dBFT-enabled node:
https://github.com/bane-labs/go-ethereum/blob/7484c42b6d4686bdcf81bdb5f9671f52567bb78f/antimev/keystore.go#L435-L438

---

> From what I see, there's a case when it will silently skip keystore initialisation if both file/password are not provided

There is a default file path, so it cannot be empty, but let me have some tests and find if we need a better check.

So these are the cases and results:
1. File doesn't exist (whether password provided or not) — `Fatal: Failed to find or load antimev keystore: no such file or directory`;
2. File exists but empty/invalid format (whether password provided or not) — `Fatal: Failed to find or load antimev keystore: unexpected end of JSON input`;
3. File is correctly provided but password not provided or wrong — `Fatal: Failed to find or load antimev keystore: could not decrypt keystore`;

However, the referred codes should get changed. Fixed.